### PR TITLE
chore: split env files

### PR DESCRIPTION
**Pull request by Siglud**

### CODE REVIEW

File: templates/scenarios/csharp/message-extension/env/.env.dev.tpl [hunk 1]
```diff
@@ -9,7 +9,4 @@ RESOURCE_SUFFIX=
 # Generated during provision, you can also add your own variables.
```
Comment (blackchoey) [inline templates/scenarios/csharp/message-extension/env/.env.dev.tpl:9]: Please reference https://github.com/OfficeDev/TeamsFx-Samples/pull/731/files to add some comments to the beginning of all the .env files (note content of every .env file is slightly different).
By the way, the `.gitignore` file also needs update to only ignore `.env.xxx.user` and `.env.local`. Above sample PR also involves the change.

File: templates/scenarios/csharp/non-sso-tab/env/.env.dev.user.tpl [hunk 2]
```diff
@@ -0,0 +1 @@
+# Secrets. Keys prefixed with `SECRET_` will be masked in Teams Toolkit logs.
```
Comment (blackchoey) [inline templates/scenarios/csharp/non-sso-tab/env/.env.dev.user.tpl:1]: If the project does not save any content to .env.xxx.user, my personal opinion is remove the `.env.xxx.user` file from template.
@jidddddd what's your opinion?

Comment (blackchoey) [reply templates/scenarios/csharp/non-sso-tab/env/.env.dev.user.tpl:1]: But we still need to ignore the .user file incase user adds it manually

File: templates/scenarios/csharp/message-extension/env/.env.dev.tpl (hunk 1 above)
Comment (Siglud) [reply templates/scenarios/csharp/message-extension/env/.env.dev.tpl:9]: Fixed

Comment (blackchoey) [reply templates/scenarios/csharp/message-extension/env/.env.dev.tpl:9]: Please also add comments to `.env.dev` and `.env.local` (note they're different)

File: templates/scenarios/csharp/non-sso-tab/env/.env.dev.user.tpl (hunk 2 above)
Comment (jidddddd) [reply templates/scenarios/csharp/non-sso-tab/env/.env.dev.user.tpl:3]: @blackchoey yes we can remove the file from template

**Review by blackchoey — APPROVED** [review]

**Pull request merged**